cmd/admin-config-set.go

```diff
@@ -81,37 +88,55 @@ func (u configSetMessage) JSON() string {
 }
 
 // checkAdminConfigSetSyntax - validate all the passed arguments
-func checkAdminConfigSetSyntax(ctx *cli.Context) {
-	if len(ctx.Args()) == 0 || len(ctx.Args()) > 2 {
+func checkAdminConfigSetSyntax(ctx *cli.Context) string {
+	if len(ctx.Args()) == 0 {
 		cli.ShowCommandHelpAndExit(ctx, "set", 1) // last argument is exit code
 	}
+	if len(ctx.Args()) == 1 {
+		return "fullSet"
+	}
+	// if 2 or more arguments are passed
+	return "partialSet"
 }
 
 // main config set function
 func mainAdminConfigSet(ctx *cli.Context) error {
+	// Initializations
+	aliasedURL := ctx.Args().Get(0)
+	isFullConfigSet := false
 
 	// Check command arguments
-	checkAdminConfigSetSyntax(ctx)
+	if checkAdminConfigSetSyntax(ctx) == "fullSet" {
+		isFullConfigSet = true
+	}
 
 	// Set color preference of command outputs
 	console.SetColor("SetConfigSuccess", color.New(color.FgGreen, color.Bold))
 	console.SetColor("SetConfigFailure", color.New(color.FgRed, color.Bold))
 
-	// Get the alias parameter from cli
-	args := ctx.Args()
-	aliasedURL := args.Get(0)
-
 	// Create a new Minio Admin Client
 	client, err := newAdminClient(aliasedURL)
 	fatalIf(err, "Cannot get a configured admin connection.")
-
-	// Call get config API
-	fatalIf(probe.NewError(client.SetConfig(os.Stdin)), "Cannot set server configuration file.")
-
-	// Print set config result
-	printMsg(configSetMessage{
-		setConfigStatus: true,
-	})
+	if isFullConfigSet {
+		// Call set config API
+		fatalIf(probe.NewError(client.SetConfig(os.Stdin)), "Cannot set server configuration file.")
+		// Print set config result
+		printMsg(configSetMessage{setConfigStatus: true})
+	} else {
+		argsMap := make(map[string]string)
+		for _, arg := range ctx.Args().Tail() {
+			argSplit := strings.SplitN(arg, "=", 2)
+			if strings.Index(arg, "{") == -1 && len(argSplit)%2 == 1 {
```

Comment:
Can you add a comment what are we doing this for ?

Reply:
OK